src/odemis/acq/align/tdct.py

```diff
@@ -0,0 +1,193 @@
+            poi: list of transformed point of interest coordinates
+                image_px: coordinates in image pixels (0, 0 top left)
+                px:  coordinates in microscope image pixels (0, 0 image center)
+                px_um: coordinates in microscope image meters (0, 0 image center)
+            transfromation:
+                rotation_eulers: transformation rotation (eulers)
+                rotation_quaternion: transformation rotation (quaternion)
+                scale: transformation scale
+                translation_around_rotation_center: transformation translation
+    """
+
+    # fib coordinates need to be x, y, z for 3DCT
+    if fib_coords.shape[-1] == 2:
+        fib_coords = numpy.column_stack((fib_coords, numpy.zeros(fib_coords.shape[0])))
+
+    # coordinates need to be float32 for 3DCT
+    fib_coords = fib_coords.astype(numpy.float32)
+    fm_coords = fm_coords.astype(numpy.float32)
+
+    # get first channel only, assume all channels are the same shape
+    if fm_image.ndim == 4:
+        fm_image = fm_image[0, :, :, :]
+    if fm_image.ndim == 5:
+        fm_image = fm_image[0, 0, :, :, :]
+
+    # get rotation center
+    halfmax_dim = int(max(fm_image.shape) * 0.5)
+    rotation_center = (halfmax_dim, halfmax_dim, halfmax_dim)
+
+    # get fib pixel size (meters)
+    fib_pixel_size = fib_image.metadata[model.MD_PIXEL_SIZE][0]
+
+    # fib image shape minus metadata, fib_pixelsize (microns), fm_image_shape
+    image_props = [fib_image.shape, fib_pixel_size * 1e6, fm_image.shape]
+
+    assert fm_coords.dtype == numpy.float32, "FM coordinates must be float32"
+    assert fib_coords.dtype == numpy.float32, "FIB coordinates must be float32"
+    assert fm_coords.shape[-1] == 3, "FM coordinates must be 3D (x, y, z)"
+    assert fib_coords.shape[-1] == 3, "FIB coordinates must be 3D (x, y, z)"
+    assert fib_coords.shape == fm_coords.shape, "FIB and FM coordinates must have the same shape"
+    assert fib_image.ndim == 2, "FIB Image must be 2D"
+    assert fm_image.ndim == 3, "FM Image must be 3D"
+    assert fib_pixel_size is not None, "FIB Pixel Size must be set"
+
+    logging.debug(
+        f"Running 3DCT correlation with FIB image shape: {fib_image.shape}, FM image shape: {fm_image.shape}"
+    )
+
+    # run correlation
+    correlation_results = run_correlation(
+        fib_coords=fib_coords,
+        fm_coords=fm_coords,
+        poi_coords=poi_coords,
+        image_props=image_props,
+        rotation_center=rotation_center,
+        path=path,
+    )
+
+    return correlation_results
+
+def get_poi_coordinate(correlation_results: dict) -> Tuple[float, float]:
+    """Get the the point of interest coordinate from correlation data
+    and convert from micrometers to meters in the microscope image coordinate system.
+    The coordinate is centred at the image centre.
+    :param correlation_results: the correlation results
+    :return: the point of interest coordinate in meters
```

Comment:
If the coordinates are in meters, with origin on the image center, then, it's quite close from the "standard" world coordinates in Odemis, which I assume was the goal of this function! *However*, in Odemis, +Y in world coordinates goes *up*. Is that the case here, or does the "poi" output follows the convention that +Y goes down? I would suggest to ensure the result follows the Odemis convention, so that it's easy to directly use (ie, please check if you need to put a `-` before the `poi_coord[1]`)

Reply:
It is actually the coordinate system used by the milling patterns, which is the same as world coordinates in odemis (by coincidence, not design). I've added the axes for the coordinates (x+ -> right, y+ -> up) in the docstring for clarity